client/src/User.re

```diff
@@ -18,30 +20,95 @@ module GetNotes = [%graphql
 
 module GetNotesComponent = ReasonApollo.CreateQuery(GetNotes);
 
-let component = ReasonReact.statelessComponent("User");
+let getNotesQuery = (~userName, ~offset=0, ()) =>
+  GetNotes.make(~userName, ~offset, ~limit=Config.sketchListLimit, ());
 
+let updateQuery = (prev, next) => {
+  /* Update Apollo Store with [@bs.raw {||}] for now, since the type comming in is a generic Js.Json.t for now*/
```

Comment:
I think with upgraded `reason-apollo` from #197, we could do improve this bit.

Perhaps we could avoid `bs.raw` totally?
```reason
external unsafeFromJson: Js.Json.t => GetNotes.t = "%identity";
external unsafeToJson: GetNotes.t => Js.Json.t = "%identity";

let updateQuery = (prev, next) => {
  let prev = unsafeFromJson(prev);
  let fetchMoreResult = ReasonApolloQuery.fetchMoreResultGet(next);

```

Reply:
No, I have implemented this thing after the update only. This query is used from their latest release example only https://github.com/apollographql/reason-apollo/blob/7eef7ddcefb7805a455905803f13889a092a3148/examples/swapi/src/Persons.re#L52-L53

Reply:
Ok! In any case, we might want to rewrite the code inside `bs.raw` to use ES5. `bsb` itself compiles to ES5.

Reply:
I've tried how this might look like without `bs.raw` below.
What do you think we should go with?

```reason
external unsafeFromJson: Js.Json.t => GetNotes.t = "%identity";
external unsafeToJson: GetNotes.t => Js.Json.t = "%identity";

let updateQuery = (prev, next) => {
  let prev = unsafeFromJson(prev);
  let fetchMoreResult = ReasonApolloQuery.fetchMoreResultGet(next);

  (
    switch (fetchMoreResult) {
    | None => prev
    | Some(moreNotes) =>
      let moreNotes = unsafeFromJson(moreNotes);
      {"note": Belt.Array.concat(prev##note, moreNotes##note)};
    }
  )
  |> unsafeToJson;
};```

Reply:
Okay i will try to re-write using reason.re or if it doesn't work i will switch it back to es5.

Reply:
We can not use reason we need spread operator under obj, because it copies the `[Symbol[id]]` of the prev object to next object. Also we can not use Js.obj.assign because it does not copy the `Symbol` key. I have changed this to the es5 syntax though.

Reply:
@nimish-gupta I'm not sure I understand. Why do we need to copy the symbol?

Reply:
@nimish-gupta Why can't we use Reason with a binding to Object.assign in this case then?

Reply:
I have tried using https://bucklescript.github.io/bucklescript/api/Js.Obj.html#VALassign but this function was also not copying the `[Symbol[id]]`. We can create a binding with `Object.assign` but then it will be same like we doing in the `es5`. I can change above code to use `Object.assign` binding.

Reply:
@nimish-gupta let's try to do it in reason.
the binding from below comment might be useful for you:
https://github.com/BuckleScript/bucklescript/pull/2774#issuecomment-434013060